deepdrivemd/data/stream/aggregator_reader.py

```diff
@@ -0,0 +1,261 @@
+import adios2
+import numpy as np
+from deepdrivemd.utils import t1Dto2D
+from pathlib import Path
+from typing import List
+from deepdrivemd.data.stream.adios_utils import AdiosStreamStepRW
+
+
+class StreamVariable:
+    """
+    This class is used to read a variable from BP file
+
+    Attributes
+    ----------
+    name : str
+         variable name in adios file
+    dtype : type
+         variable type, for example, np.uint8
+    structure : int
```

Comment:
Can we update `structure` to be type str? It would make the code more readable, otherwise you have to look back at this class to know what the 0,1,2 refers to. Perhaps we could do "array", "scalar", "str".

Reply:
Converted to using enumerations.